packages/my-near-wallet/src/lib/my-near-wallet.ts

```diff
@@ -224,6 +224,18 @@ const MyNearWallet: WalletBehaviourFactory<
       });
     },
 
+    async signMessage({ message, nonce, recipient, callbackUrl, state }) {
+      logger.log("MyNearWallet:signMessage", {
+        message,
+        nonce,
+        recipient,
+        callbackUrl,
+        state,
+      });
+
+      throw new Error(`Method not supported by ${metadata.name}`);
+    },
+
```

Comment:
I don't know why this wouldn't be supported by `MyNearWallet`/`NearWallet` since all we need to sign a message is available without needing to redirect to the wallet website. This is how it could be done:
```typescript
const { contract } = store.getState();

if (!_state.wallet.isSignedIn() || !contract) {
  throw new Error("Wallet not signed in");
}

const account = _state.wallet.account();

const { networkId } = account.connection;
const accountId = account.accountId;

const keyPair = await _state.keyStore.getKey(networkId, accountId);
const signer = await nearAPI.InMemorySigner.fromKeyPair(
  networkId,
  accountId,
  keyPair
);

return signer.signMessage(new Uint8Array(Buffer.from(message)));
```

Reply:
That would sign a message using the function call access key which was added on the Dapp side. I think this API is meant to sign messages with a full access key, which is only available on the wallet side.

Reply:
Which makes sense- because otherwise you're handing DApps "signing power" when you sign into them. Only Full Access keys should be allowed to be the source of a wallet account's signed message.